### PR TITLE
Async compatible `SQLPanel`

### PR DESCRIPTION
**Pull request by salty-ivy**
# Description

By instrumenting  the `SQLPanel` in `sync_to_async` we can monkey patch all the connection in a dedicated thread and store those connections there meanwhile all cross compatible views and operations in async context will take place in that threaded thus they will use those patched db connections which will help us store all async and sync operations.

more explanation of `sync_to_async`: https://docs.djangoproject.com/en/4.2/topics/async/#sync-to-async

Fixes https://github.com/jazzband/django-debug-toolbar/issues/1430
Fixes  https://github.com/jazzband/django-debug-toolbar/issues/932

# Checklist:

- [x] I have added the relevant tests for this change.
- [x] I have added an item to the Pending section of ``docs/changes.rst``.

### CODE REVIEW

**Comment by salty-ivy** [comment]
@tim-schilling  I think this is ready as per the discussion tests have been covered by #1835

File: debug_toolbar/middleware.py [hunk 1]
```diff
@@ -112,7 +112,10 @@ async def __acall__(self, request):
 
         # Activate instrumentation ie. monkey-patch.
         for panel in toolbar.enabled_panels:
-            panel.enable_instrumentation()
+            if panel.panel_id == "SQLPanel":
+                await sync_to_async(panel.enable_instrumentation)()
```
Comment (tim-schilling) [inline debug_toolbar/middleware.py:116]: Is it possible to do this within the `SQLPanel.enable_instrumentation` itself?

Comment (tim-schilling) [reply debug_toolbar/middleware.py:116]: It's a bit odd to see specific logic for a single panel in the middleware. It'll be easier to understand the middleware if we don't have specific panel logic within it. It's also easier to explain why we do that in the panel and have it be in the place a person is most likely to be looking for it.

Comment (tim-schilling) [reply debug_toolbar/middleware.py:116]: Oh, I like that.

Comment (salty-ivy) [reply debug_toolbar/middleware.py:116]: we only have one panel as of now that would utilise this pattern, though it seems really intuitive.

Comment (elineda) [reply debug_toolbar/middleware.py:116]: yep but if a people create a third party panel he need to change the code of the core for make work. That's not a great thing.

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -6,7 +6,9 @@ Pympler
 Roboto
 Transifex
 Werkzeug
+aenable_instrumentation
```
Comment (tim-schilling) [inline docs/spelling_wordlist.txt:9]:
```suggestion
aenable
```

File: docs/spelling_wordlist.txt [hunk 3]
```diff
@@ -15,6 +17,7 @@ checkbox
 contrib
 dicts
 django
+enable_instrumentation
```
Comment (tim-schilling) [inline docs/spelling_wordlist.txt:20]:
```suggestion
```

File: debug_toolbar/panels/__init__.py [hunk 4]
```diff
@@ -154,6 +154,9 @@ def enable_instrumentation(self):
 
         Unless the toolbar or this panel is disabled, this method will be
         called early in ``DebugToolbarMiddleware``. It should be idempotent.
+
+        Add aenable_instrumentation method to the panel class to support async
+        instrumentation logic.
```
Comment (elineda) [inline debug_toolbar/panels/__init__.py:159]: maybe an "if needed" ? it's not mandatory for async to have a aenable ?

**Comment by salty-ivy** [comment]
@tim-schilling  Apparently `instrumentation` is causing the lint fail, even if I have added it to word list.

File: debug_toolbar/panels/__init__.py (hunk 4 above)
Comment (salty-ivy) [reply debug_toolbar/panels/__init__.py]: yup, although `async instrumentation logic` itself means that `await` is required in that case `aenable_instrumentation` would be the only choice?

**Review by tim-schilling — APPROVED** [review]

**Pull request merged**